### PR TITLE
Makes it submit solutions with obj(all)<0 when obj(touched)>0.

### LINKED ISSUE #82: Fails to find solution.
This instance has a solution, but the solver doesn't find it in "find-best-pair" mode. It does find it when the two tokens are explicitly passed as parameters.
[instance_5342282_2020-10-14T14_15_30.328294632+00_00.json.txt](https://github.com/gnosis/dex-open-solver/files/5379360/instance_5342282_2020-10-14T14_15_30.328294632%2B00_00.json.txt)

### PR DESCRIPTION
**Pull request by marcovc**
The previous behaviour of the open solver was to never submit solutions when u-du over
all orders is negative. This PR relaxes this constraint - now it can submit a solution
with negative u-du over all orders if u-du is positive over touched orders.

Closes #82.

### CODE REVIEW

File: dex_open_solver/best_token_pair_solver/solver.py [hunk 1]
```diff
@@ -51,17 +51,28 @@ def match_token_pair(token_pair, accounts, orders, fee):
...
+    if touched_only:
+        all_orders = orders
+        orders = [o for o in orders if o.buy_amount > 0]
+
     # Compute objective value for current token pair solution.
     objective = compute_objective(prices, accounts_updated, orders, fee)
 
+    if touched_only:
+        orders = all_orders
+
```
Comment (fleupold) [inline dex_open_solver/best_token_pair_solver/solver.py:75]: ubermegananonit: I would try to only use one if statement here, but that's just personal preference.

```python
if touched_only:
  touched = [o for o in orders if o.buy_amount > 0]
  objective = compute_objective(prices, accounts_updated, touched, fee)
else:
  objective = compute_objective(prices, accounts_updated, orders, fee)
```

**Review by fleupold — APPROVED** [review]
Nice detective work and a straightforward change!

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2F42wQXwITfQbDGKqUP7%2Fgiphy.gif&token=detective" data-gitmeme-token="detective"></a>

File: dex_open_solver/best_token_pair_solver/solver.py (hunk 1 above)
Comment (marcovc) [reply dex_open_solver/best_token_pair_solver/solver.py]: Thanks, this is better.

**Review by josojo — APPROVED** [review]
Good catch!

**Pull request merged**